### PR TITLE
DM-39173: Update the API for accessing the image, mask, and variance planes

### PR DESCRIPTION
**Pull request by isullivan**
Also fixes a major bug in the dipole calculation that was due to the old API.

### CODE REVIEW

**Review by BrunoSanchez — APPROVED** [review]
This all looks good to me.

File: python/lsst/ip/diffim/dipoleFitTask.py [hunk 1]
```diff
@@ -860,7 +860,7 @@ def fitDipole(self, source, tol=1e-7, rel_weight=0.1,
         def computeSumVariance(exposure, footprint):
             box = footprint.getBBox()
             subim = afwImage.MaskedImageF(exposure.getMaskedImage(), box, origin=afwImage.PARENT)
-            return np.sqrt(np.nansum(subim.getArrays()[1][:, :]))
+            return np.sqrt(np.nansum(subim.getArrays()[2][:, :]))
```
[the remark this answers is not in the document]
Comment (TallJimbo) [reply python/lsst/ip/diffim/dipoleFitTask.py]: I think
```
return np.sqrt(np.nansum(subim.variance.array[:, :]))
```
is what you want.

Though (and this is, I think, out of scope), it's a bit strange that this function seems to be pretending to accept a footprint and then ignoring that footprint in favor of its bounding box.  If it's just going to the bounding box, it should probably just take a box.

[the remark this answers is not in the document]
Comment (isullivan) [reply python/lsst/ip/diffim/dipoleFitTask.py]: Yes, the API is fixed on the next commit. I just separated the bug fix from the API change.

[the remark this answers is not in the document]
Comment (TallJimbo) [reply python/lsst/ip/diffim/dipoleFitTask.py]: Ah, sorry for the noise.  I didn't see the context and misinterpreted @BrunoSanchez 's comment.

[the remark this answers is not in the document]
Comment (BrunoSanchez) [reply python/lsst/ip/diffim/dipoleFitTask.py]: Sorry for the noise as well. I took the chance to note this in github after a chat I had with Ian privately about it. Never thought someone would be looking at this right now...

**Pull request merged**